compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/CodeAnalyzer.java

```diff
@@ -1343,6 +1344,20 @@ private void checkForExportableType(BTypeSymbol symbol, DiagnosticPos pos) {
         }
     }
 
+    public void visit(BLangLetExpression letExpression) {
+        int ownerSymTag = this.env.scope.owner.tag;
+        if ((ownerSymTag & SymTag.RECORD) == SymTag.RECORD) {
+            dlog.error(letExpression.pos, DiagnosticCode.LET_EXPRESSION_NOT_YET_SUPPORTED_RECORD_FIELD);
```

Comment:
Shall we create an issue for these if we haven't already?

Reply:
Please refer to the issue. 
https://github.com/ballerina-platform/ballerina-lang/issues/21382